logdevice/common/test/NodesConfigurationPollerTest.cpp

```diff
@@ -0,0 +1,116 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc. and its affiliates.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+#include "logdevice/common/NodesConfigurationPoller.h"
+
+#include <algorithm>
+
+#include <folly/Random.h>
+#include <gtest/gtest.h>
+
+#include "logdevice/common/ObjectPoller.h"
+#include "logdevice/common/RandomNodeSelector.h"
+#include "logdevice/common/configuration/Configuration.h"
+#include "logdevice/common/configuration/nodes/NodesConfigurationCodec.h"
+#include "logdevice/common/test/TestUtil.h"
+#include "logdevice/common/toString.h"
+
+using namespace facebook::logdevice;
+
+namespace {
+
+class NodesConfigurationPollerExtended : public NodesConfigurationPoller {
+ public:
+  static std::vector<node_index_t> nodesCalled;
+  virtual ~NodesConfigurationPollerExtended() {}
+
+  NodesConfigurationPollerExtended(
+      Poller::Options options,
+      VersionExtFn version_fn,
+      Callback cb,
+      folly::Optional<u_int32_t> node_order_seed,
+      folly::Optional<Version> conditional_base_version)
+      : NodesConfigurationPoller(std::move(options),
+                                 std::move(version_fn),
+                                 std::move(cb),
+                                 node_order_seed,
+                                 std::move(conditional_base_version)) {}
+
+ protected:
+  Poller::RequestResult sendRequestToNode(Poller::RoundID round,
+                                          node_index_t node) {
+    /* nodesCalled.push_back(node); */
+    return Poller::RequestResult::OK;
+  }
+
+  ClusterState* getClusterState() {
+    return nullptr;
+  }
+
+  std::shared_ptr<const configuration::nodes::NodesConfiguration>
+  getNodesConfiguration() const {
+    auto node_config = createSimpleNodesConfig(1 /* 1 node */);
+    auto server_config = ServerConfig::fromDataTest(
+        "random_node_selector_test", std::move(node_config));
+
+    auto nodes = server_config->getNodes();
+    auto node_id =
+        NodeID(nodes.begin()->first, nodes.begin()->second.generation);
+    return server_config->getNodesConfigurationFromServerConfigSource();
```

Comment:
I don't understand what's happening here. Why not just `return createSimpleNodesConfig(1 /* 1 node */)`?

Reply:
Trying that throws error:
`/data/LogDevice/logdevice/common/test/NodesConfigurationPollerTest.cpp: In member function ‘virtual std::shared_ptr<const facebook::logdevice::configuration::nodes::NodesConfiguration> {anonymous}::NodesConfigurationPollerExtended::getNodesConfiguration() const’:
/data/LogDevice/logdevice/common/test/NodesConfigurationPollerTest.cpp:58:35: error: could not convert ‘facebook::logdevice::createSimpleNodesConfig(size_t)()’ from ‘facebook::logdevice::ServerConfig::NodesConfig {aka facebook::logdevice::configuration::NodesConfig}’ to ‘std::shared_ptr<const facebook::logdevice::configuration::nodes::NodesConfiguration>’`

Reply:
Oh that's because `createSimpleNodesConfig` returns the old NodesConfig. You can use utils from `NodesConfigurationTestUtil.h` to create a dummy NodesConfiguration.